### DISCUSSION TITLE
Installing non conda software

### DISCUSSION THREAD

**Discussion by s-canchi** [nih-cfde/training-and-engagement#369] [Conda Q&A] [answered]
Sometimes I come across instructions to install software using a different package manager, presumably because the software is not available through conda. Can software that is not downloaded through conda be installed in a conda environment to keep it isolated? Or is there a better way to do this?

**Comment by s-canchi** [comment, marked as answer]
- Yes! That is another way to use conda. Any software installed with conda or otherwise ends up in the environment folder which is isolated.
- More information on installing non-conda programs and such —> https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-pkgs.html#installing-similar-packages

**Reply by ctb** [reply]
>> Can software that is not downloaded through conda be installed in a conda environment to keep it isolated? Or is there a better way to do this?
>Yes! That is another way to use conda. Any software installed with conda or otherwise ends up in the environment folder which is isolated.

This may or not be correct - in general, it is not, unless you take care to build and install the software within the conda environment. Only packages that are installed into the bin/lib/etc subdirectories of a particular environment work within that environment.

In particular,
* commercially installed software will generally go in system directories, and will not be managed by conda in any way
* packages installed via homebrew, debian apt, redhat yum, etc will generally go in directories managed by those packaging systems, and will not be managed by conda in any way
* packages installed via python pip and other systems where conda can install the package installer (ruby gems, js npm) will be installed under the activate conda environment

so... it's complicated :)
